app/config/windows.js

```diff
@@ -8,11 +9,28 @@ const cfg = new Config({
   }
 });
 
+function validateAndFixWindowPosition(position, size) {
+  const displays = electron.screen.getAllDisplays();
+  const [x, y] = position;
+  const positionIsValid = displays.some(({workArea}) => {
+    return x >= workArea.x && x <= workArea.x + workArea.width && y >= workArea.y && y <= workArea.y + workArea.height;
```

Comment:
We should test if size is valid too. We don't want to have a valid position and an almost invisible window. For example if `x === workArea.width - 1` or `y === workArea.height - 1`.

Reply:
I had contemplated this, but decided to keep it simple and follow YAGNI. If we were to get into this case, the use would have had to positioned the window there themselves and if it's on the screen the standard <kbd>ctrl</kbd> + <kbd>arrow</kbd> shortcuts would work.

If we go with this method we would need to define how big a work area is permissible. We could perhaps change the logic to check the center of the window, if the center of a terminal window is in the workarea, then we're good, if not, relocate? (This would cover both the edge case you mention, and the original "my monitor is no longer plugged in" case)

Reply:
I think that considering window center is really better than top-left corner.
I know there is some shortcuts (different on each OS), but I am pretty sure that some users know nothing about them.

I will fix this by myself, because this PR had a too long wait time. Sorry again for this

Reply:
No need to consider new window center because startX/startY is already corrected by existing code.